src/main/resources/drivers/CiscoNext-GenSeries.js

```diff
@@ -0,0 +1,384 @@
+      { prompt: new RegExp(ANSI + ".*[>#]\\s?$", "m") }
+    ],
+    macros: {
+      enable: { options: ["enable"], target: "enable" }
+    }
+  },
+
+  exec: {
+    prompt: PROMPT_EXEC,
+    pager: {
+      match: PAGER_RE,
+      response: " "
+    }
+  },
+
+  enable: {
+    prompt: PROMPT_ENABLE,
+    commands: [
+      { command: "enable", expect: /Password:\s*$/mi, response: "%enablePassword%", optional: true },
+      { command: "terminal datadump", optional: true },
+      { command: "terminal length 0", optional: true },
+      { command: "terminal width 511", optional: true }
+    ],
+    pager: {
+      match: PAGER_RE,
+      response: " "
+    }
+  }
+};
+
+/* ========================= HELPERS ========================= */
+
+function toCiscoDottedMac(s) {
+  if (!s) return s;
+  var hex = String(s).toLowerCase().replace(/[^0-9a-f]/g, "");
+  if (hex.length !== 12) return s;
+  return hex.slice(0,4) + "." + hex.slice(4,8) + "." + hex.slice(8,12);
+}
+
+function ensureIfaceShape(obj, fallbackName, isL3) {
+  if (!obj || typeof obj !== "object") obj = {};
+  obj.name = (obj.name != null ? String(obj.name) : String(fallbackName || "")).trim();
+  if (!obj.name) return null;
+
+  if (obj.description == null) obj.description = "";
+  if (obj.level3 == null) obj.level3 = !!isL3;
+
+  if (obj.enabled == null) obj.enabled = false;
+  if (obj.adminStatus == null) obj.adminStatus = obj.enabled ? "Up" : "Down";
+  if (obj.linkStatus == null) obj.linkStatus = obj.adminStatus;
+
+  if (!Array.isArray(obj.ip)) obj.ip = [];
+  obj.ip = obj.ip.filter(function (x) { return x && typeof x === "object" && (x.ip || x.address); });
+
+  if (obj.level3) {
+    if (obj.vrf == null) obj.vrf = "default";
+  } else {
+    if (obj.vrf != null) delete obj.vrf;
+  }
+
+  if (typeof obj.mac === "string") obj.mac = toCiscoDottedMac(obj.mac);
+  if (!obj.mac) delete obj.mac;
+
+  if (typeof obj.macAddress === "string") obj.macAddress = toCiscoDottedMac(obj.macAddress);
+  if (!obj.macAddress) delete obj.macAddress;
+
+  return obj;
+}
+
+function scrubMacs(o) {
+  if (!o) return o;
+  if (!o.mac) delete o.mac;
+  if (!o.macAddress) delete o.macAddress;
+  return o;
+}
+
+// Dedup guards
+var addedIfaces = {};
+function addIfaceOnce(device, kindObj) {
+  if (!kindObj || !kindObj.name) return;
+  var key = String(kindObj.name).toLowerCase();
+  if (addedIfaces[key]) return;
+  addedIfaces[key] = true;
+  device.add("networkInterface", kindObj);
+}
+
+var addedModules = {};
+function addModuleOnce(device, mod) {
+  if (!mod) return;
+  // Build a stable dedup key: slot|partNumber|serialNumber (lower-cased)
+  var k = [
+    mod.slot || "",
+    mod.partNumber || "",
+    mod.serialNumber || ""
+  ].join("|").toLowerCase();
+  if (addedModules[k]) return;
+  addedModules[k] = true;
+  device.add("module", mod);
+}
+
+/* ========================= SNAPSHOT ========================= */
+
+function snapshot(cli, device, config) {
+  try { cli.command(""); } catch (e) {}
```

Comment:
The `snapshot` function is called with CLI starting in `ssh` mode. You should call `cli.macro("enable");` here to switch from `ssh` mode to `enable` mode using possible transient modes.